WordPress/wordcamp.org
# Remote CSS: Migrate to using a GitHub app for authentication

GitHub is[ deprecating query parameters for authentication](https://developer.github.com/changes/2019-11-05-deprecated-passwords-and-authorizations-api/#authenticating-using-query-parameters) so this PR converts it to using a GitHub App instead.

This is based off the work I did for integrating [GitHub PRs on Trac](https://meta.trac.wordpress.org/changeset/9564) as it mostly re-used the basic structure to make requests with a higher rate limit.

This has also increased the rate limit from 5k requests/hour to 12.5k/hour. I'm sure we'd never have been nearing the previous rate limit.

A new GitHub App was created and the required secrets added to WordCamp.org already.
Once merged, the old secrets (`REMOTE_CSS_GITHUB_ID`/`REMOTE_CSS_GITHUB_SECRET`) can be removed.

### How to test the changes in this Pull Request:

1. Apply PR
2. Open a test site with Remote CSS enabled, hit the Update CSS button on `/wp-admin/themes.php?page=remote-css` verify that it shows a success.
3. If wanted, verify that the Response Headers from GitHub indicate the higher rate limit, add `var_dump( $preempt['headers'] );` as line 104 of `platforms/github.php` and observe the value of the `x-ratelimit-limit` header.
